src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -418,13 +428,21 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
                 $markingStoreDefinition = new Reference($workflow['marking_store']['service']);
             }
 
-            $workflowDefinition = new DefinitionDecorator('workflow.abstract');
+            $definitionDefinition->addTag('workflow.definition', array(
+                'name' => $name,
+                'type' => $type,
+                'marking_store' => isset($workflow['marking_store']['type']) ? $workflow['marking_store']['type'] : null,
+            ));
+            $definitionDefinition->setPublic(false);
```

Comment:
Actually, Having the definition public is a an issue, because it's mutable. So people could change the definition at runtime and cause lot of trouble.
Anyway, I don't have a better way to do it. Or may be with the reflection... WTDY?